### ISSUE TITLE
Best practice for enhancement(s)

### ISSUE THREAD

**Issue by paskainos** [danmindru/responsive-html-email-signature#57]
I’ve added [clipboard.js](https://clipboardjs.com/) support, and added [contenteditable](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable) to certain elements. I used ```npm install clipboard --save``` (per [guidance](https://clipboardjs.com/#install)) to install clipboard.js, but I have a ‘utils.js’ file (linked in document ```head```), residing outside the root directory, which:

1. makes certain elements editable…
1. removes ‘contentEditable’ attribute before copy…
1. copies text (html)…
1. reinstates contentEditable after copy…
1. displays ```alert("Copied!")```

In an effort to make this available and reusable, I’m wondering where to include the utils.js file (code). Any guidance is appreciated.

**Comment by danmindru** [comment]
Hi @paskainos, that sounds quite interesting. Some sort of interactive view I reckon? 

In my mind it should work like an enhancement that starts up a small server in parallel with the gulp task.
The server returns an HTML file with util.js included and loads up the email templates in some sort of view (more on this below).
This way the 'legacy' build way would still work like it does, so people can use it as it is or in the so called 'interactive mode'.  It's also a good idea to not mix things up, methinks :-) 

This would allow for more enhancements later, like:
- [ ] viewing one or more templates at a time
- [ ] showing a preview of desktop, tablet & mobile side-by-side in the same view
- [ ] ...even adding the ability to generate configs back to save the edits

Does that make sense?
Implementation-wise, this server doesn't have to do much except for serving a html file and watching for template re-generation. There are a lot of solutions here:
- [ ] gulp-based (new task with browser sync for updates)
- [ ] rollup or webpack based (they'll come with their own watch/server tech)
- [ ] npm scripts with some sort of light server, like [http-server](https://www.npmjs.com/package/http-server)

Let me know if this makes sense or if you have any other questions :-)
I can also go into detail with some of the options if you'd like.
